lib/windows.js

```diff
@@ -93,35 +136,47 @@ function windows() {
 	// The text as a JavaScript string
 	const windowTitle = wchar.toString(windowTextBufferClean);
 
-	// Allocate a buffer to store the process ID
-	const processIdBuffer = ref.alloc('uint32');
-	// Write the process ID creating the window to the buffer (it returns the thread ID, but it's not used here)
-	user32.GetWindowThreadProcessId(activeWindowHandle, processIdBuffer);
-	// Get the process ID as a number from the buffer
-	const processId = ref.get(processIdBuffer);
-	// Get a "handle" of the process
-	const processHandle = kernel32.OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, false, processId);
+	// Return the Process ID & Process Handle from the Active Window Handle
+	const [processId, processHandle] = getProcessIdAndHandle(activeWindowHandle);
 
 	if (ref.isNull(processHandle)) {
 		return undefined; // Failed to get process handle
 	}
 
-	// Set the path length to more than the Windows extended-length MAX_PATH length
-	const pathLengthBytes = 66000;
-	// Path length in "characters"
-	const pathLengthChars = Math.floor(pathLengthBytes / 2);
-	// Allocate a buffer to store the path of the process
-	const processFileNameBuffer = Buffer.alloc(pathLengthBytes);
-	// Create a buffer containing the allocated size for the path, as a buffer as it must be writable
-	const processFileNameSizeBuffer = ref.alloc('uint32', pathLengthChars);
-	// Write process file path to buffer
-	kernel32.QueryFullProcessImageNameW(processHandle, 0, processFileNameBuffer, processFileNameSizeBuffer);
-	// Remove null characters from buffer
-	const processFileNameBufferClean = ref.reinterpretUntilZeros(processFileNameBuffer, wchar.size);
-	// Get process file path as a string
-	const processPath = wchar.toString(processFileNameBufferClean);
-	// Get process file name from path
-	const processName = path.basename(processPath);
+	// Return the Process Path & Process Name from a Process Handle
+	let [processPath, processName] = getProcessPathAndName(processHandle);
+
+	let newProcessName;
+	let newProcessPath;
+
+	// ApplicationFrameHost & Universal Windows Platform Support
+	if (processName === 'ApplicationFrameHost.exe') {
+
+		const windowProcess = ffi.Callback('bool', ['pointer', 'int32'], (hwnd, lParam) => {
+
+			// Return the Process ID & Process Handle from the Active Window Handle
+			const [processId, processHandle] = getProcessIdAndHandle(hwnd);
```

Comment:
The `processId` here is not actually used anywhere.

Reply:
Process ID is used to get to the handle. I'd rather not duplicate code in the base, because even if I separated it, I'd need a "getProcessId" function, then a "getProcessHandle" function, which is essentially the same thing, except I have to write the same logic twice in the code (once when requesting from the parent window, and once from the sub).

Reply:
Leaving open for feedback.

Reply:
I meant it should be `const [, processHandle] = getProcessIdAndHandle(hwnd);`